### PR TITLE
Fix undefined method `symbolize_keys` ActionController::Parameters (part 2)

### PR DESCRIPTION
**Pull request by josemigallas**
Follows up on https://github.com/3scale/porta/pull/2898

* Checks whether payment_gateway_settings is already a Hash, otherwise `permit` will raise an error.
* Implements some needed tests for this flow that where actually disabled 🙄 causing us to slip this error in the first place

```
Started DELETE "/admin/api/accounts/34.json?access_token=thetoken" for 10.0.144.73 at 2022-02-04 15:09:15 +0000
Processing by Admin::Api::AccountsController#destroy as JSON
  Parameters: {"access_token"=>"thetoken", "id"=>"34"}
PermissionEnforcer: level = rw
[ActiveJob] Enqueued ActiveJob::QueueAdapters::SidekiqAdapter::JobWrapper#1cf591ff8394aec8c980327b with args: [{"job_class"=>"SphinxAccountIndexationWorker", "job_id"=>"265fdcd5-3a89-47a7-bfad-485b41e9f901", "provider_job_id"=>nil, "queue_name"=>"default", "priority"=>nil, "arguments"=>[{"_aj_globalid"=>"gid://system/Account/34"}], "executions"=>0, "locale"=>"en"}]
[ActiveJob] Enqueued SphinxAccountIndexationWorker (Job ID: 265fdcd5-3a89-47a7-bfad-485b41e9f901) to Sidekiq(default) with arguments: #<GlobalID:0x000000000dbf6d70 @uri=#<URI::GID gid://system/Account/34>>
[EventBroker] notifying subscribers of Applications::ApplicationDeletedEvent f379491b-a7d9-4b8e-9775-83a32d431321
[EventBroker] notifying subscribers of ZyncEvent 64fa78c1-2612-4921-bf32-ef3c870cad4a
PermissionEnforcer#ensure clear level
PermissionEnforcer: level =
Completed 500 Internal Server Error in 151ms (ActiveRecord: 69.3ms)

NoMethodError (undefined method `symbolize_keys' for #<ActionController::Parameters:0x000000000e003c38>):

app/models/payment_gateway_setting.rb:21:in `symbolized_settings'
app/models/account/gateway.rb:54:in `payment_gateway_options'
app/models/account/gateway.rb:20:in `payment_gateway'
app/models/account/gateway.rb:65:in `provider_payment_gateway'
app/models/account/credit_card.rb:105:in `unstore_credit_card!'
lib/deadlock_retry.rb:52:in `transaction'
app/controllers/admin/api/accounts_controller.rb:135:in `block in destroy'
lib/deadlock_retry.rb:52:in `transaction'
app/controllers/admin/api/accounts_controller.rb:134:in `destroy'
app/lib/api_authentication/by_access_token.rb:162:in `enforce'
app/lib/api_authentication/by_access_token.rb:112:in `enforce_access_token_permission'
app/controllers/admin/api/base_controller.rb:42:in `block in notification_center'
lib/notification_center.rb:30:in `silent_about'
app/controllers/application_controller.rb:140:in `silent_about'
app/controllers/admin/api/base_controller.rb:41:in `notification_center'
lib/three_scale/middleware/multitenant.rb:113:in `_call'
lib/three_scale/middleware/multitenant.rb:108:in `call'
lib/three_scale/middleware/cors.rb:21:in `call'
```

### CODE REVIEW

File: app/models/payment_gateway_setting.rb [hunk 1]
```diff
@@ -18,9 +18,11 @@ class PaymentGatewaySetting < ApplicationRecord
   # In our DB :test is stored for some providers
 
   def symbolized_settings
-    keys = PaymentGateway::GATEWAYS.map(&:fields).map(&:keys).flatten.uniq
+    settings = gateway_settings
+    return settings.symbolize_keys if settings.is_a? Hash
```
Comment (akostadinov) [inline app/models/payment_gateway_setting.rb:22]:
```suggestion
    return settings.symbolize_keys if settings.respond_to? :symbolize_keys
```
Better use duck typing here.

File: features/step_definitions/confirm_dialog_steps.rb [hunk 2]
```diff
@@ -1,10 +1,30 @@
...
+      end
+    end
+    wait_for_requests
   end
 end
+
+# should not work if it relies on some JS
+def rack_test?
+  %I[webkit selenium webkit_debug headless_chrome chrome headless_firefox firefox].exclude? Capybara.current_driver
+end
```
Comment (akostadinov) [inline features/step_definitions/confirm_dialog_steps.rb:30]: shouldn't this method go to some helper? Or to some file inside `support/` directory?

File: app/models/payment_gateway_setting.rb [hunk 3]
```diff
@@ -18,9 +18,11 @@ class PaymentGatewaySetting < ApplicationRecord
...
 
-    gateway_settings.permit(keys).to_h.symbolize_keys.except(:test)
+    keys = PaymentGateway.all.map(&:fields).map(&:keys).flatten.uniq
+    settings.permit(keys).to_h.symbolize_keys.except(:test)
```
Comment (akostadinov) [inline app/models/payment_gateway_setting.rb:25]: On line 18 there is a comment `In our DB :test is stored for some providers`. So perhaps we need to keep `:test`? I'm not sure, just a question.

**Review by akostadinov — CHANGES_REQUESTED** [review]

File: app/models/payment_gateway_setting.rb (hunk 3 above)
Comment (josemigallas) [reply app/models/payment_gateway_setting.rb:25]: I don't know either but I'd say let's keep the logic as it was (with .except)

File: features/step_definitions/confirm_dialog_steps.rb (hunk 2 above)
Comment (josemigallas) [reply features/step_definitions/confirm_dialog_steps.rb:30]: Makes sense

File: app/models/payment_gateway_setting.rb (hunk 3 above)
Comment (akostadinov) [reply app/models/payment_gateway_setting.rb:25]: Didn't notice that old version was same :)

**Review by akostadinov — APPROVED** [review]

**Pull request merged**